src/nix/flake-lock.md

```diff
@@ -2,37 +2,36 @@ R""(
 
 # Examples
 
-* Update the `nixpkgs` and `nix` inputs of the flake in the current
-  directory:
+* Create the lock file for the flake in the current directory:
 
   ```console
-  # nix flake lock --update-input nixpkgs --update-input nix
-  * Updated 'nix': 'github:NixOS/nix/9fab14adbc3810d5cc1f88672fde1eee4358405c' -> 'github:NixOS/nix/8927cba62f5afb33b01016d5c4f7f8b7d0adde3c'
-  * Updated 'nixpkgs': 'github:NixOS/nixpkgs/3d2d8f281a27d466fa54b469b5993f7dde198375' -> 'github:NixOS/nixpkgs/a3a3dda3bacf61e8a39258a0ed9c924eeca8e293'
+  # nix flake lock
+  warning: creating lock file '/home/myself/repos/testflake/flake.lock':
+  • Added input 'nix':
+      'github:NixOS/nix/9fab14adbc3810d5cc1f88672fde1eee4358405c' (2023-06-28)
+  • Added input 'nixpkgs':
+      'github:NixOS/nixpkgs/3d2d8f281a27d466fa54b469b5993f7dde198375' (2023-06-30)
   ```
 
-# Description
-
-This command updates the lock file of a flake (`flake.lock`) so that
-it contains a lock for every flake input specified in
-`flake.nix`. Existing lock file entries are not updated unless
-required by a flag such as `--update-input`.
+* Add missing inputs to the lock file for a flake in a different directory:
 
-Note that every command that operates on a flake will also update the
-lock file if needed, and supports the same flags. Therefore,
+  ```console
+  # nix flake lock ~/repos/another
+  warning: updating lock file '/home/myself/repos/another/flake.lock':
+  • Added input 'nixpkgs':
+      'github:NixOS/nixpkgs/3d2d8f281a27d466fa54b469b5993f7dde198375' (2023-06-30)
 
-```console
-# nix flake lock --update-input nixpkgs
-# nix build
-```
+  Note: When trying to refer to a flake in a direct descendant directory, write `./another`
+  instead of `another`, else Nix will try look up the flake in the registry.
```

Comment:
```suggestion
```

Maybe this isn't explicit enough in the flakeref description, but in that case it should be explicited there, not here

Reply:
Hmm I disagree. Normally when specifying a flake like with `nix develop`, `nix run` or `nix shell`, it is expected that the registry will be used. But `flake lock` only makes sense when used on paths, so the fact that nix will still try to resolve from the registry and then complain that it didn't find anything might be very surprising.

The better solution here would be to only allow a path as an input, but I didn't really see how to I could do that nicely, so I opted for this note as a bandaid for now.

Reply:
As this is a docs topic, @fricklerhandwerk, do you have an opinion on this?

Reply:
Sorry not this time, I'd have to dig into flakes documentation a lot more to judge.

Reply:
In addition to the notes here, I also opened [this PR](https://github.com/NixOS/nix/pull/9237) that should make this clearer in the flakeref docs

Reply:
Fair enough. Let's keep it that way (and thanks for the side docs PR)